api/desecapi/tests/base.py

```diff
@@ -467,18 +474,24 @@ def request_pdns_update_catalog(cls):
             'priority': 1,  # avoid collision with DELETE zones/(?P<id>[^/]+)$ (httpretty does not match the method)
         }
 
-    def assertPdnsRequests(self, *expected_requests, expect_order=True):
+    def __init__(self, methodName: str = ...) -> None:
+        super().__init__(methodName)
+        self._mock_replication = None
+
```

Comment:
that would change that to be [class attribute instead of an instance attribute](https://dzone.com/articles/python-class-attributes-vs-instance-attributes). I want it to be an instance attribute

Reply:
The following is an "academic assessment" because this attribute stuff is interesting. I have not stakes about how it's done.

The article you linked has a section called "Class Attributes Mutate to Be Instance Attributes" which says that class attributes become instance attributes when assigned inside a method. (Technically, the attribute is added to the instance attribute dict which shadows the class attribute dict.)

So, when you assign to the attribute in `setUp`, you have an instance attribute -- even when initializing it as a class attribute. Thus, your goal is achieved. :)

Now, I don't know if there's a convention if "placeholder variables" are better declared in `__init__` or on the class level. You certainly could do without declaring it at all, as the assignment in `setUp` serves just as well as the one in `__init__`. Having said that, another option would be adding an [instance variable annotation without default](https://www.python.org/dev/peps/pep-0526/#class-and-instance-variable-annotations) (`_mock_replication: None`).

Reply:
Um, the code style I followed thus far is to declare instance attributes during `__init__`, so I prefer to keep it this way. Gotta read up on instance vs class attributes though ...